### PR TITLE
feat(IMangrove): Ensure IMangrove and Mangrove align on signatures

### PR DESCRIPTION
**Pull request by lnist**
See individual commits. The last one may be skipped without much hurt.

### CODE REVIEW

File: src/MgvRoot.sol [hunk 1]
```diff
@@ -138,12 +138,12 @@ contract MgvRoot is HasMgvEvents {
   function configInfo(OLKey memory olKey)
     external
     view
-    returns (MgvStructs.GlobalUnpacked memory global, MgvStructs.LocalUnpacked memory local)
+    returns (MgvStructs.GlobalUnpacked memory _global, MgvStructs.LocalUnpacked memory _local)
```
Comment (espendk) [inline src/MgvRoot.sol:141]: This looks weird - why not avoid the `_` ?

File: src/IMangrove.sol [hunk 2]
```diff
@@ -8,70 +8,51 @@ pragma solidity >=0.7.0 <0.9.0;
...
+  function config(OLKey memory olKey)
+    external
+    view
+    returns (MgvStructs.GlobalPacked _global, MgvStructs.LocalPacked _local);
 
   function configInfo(OLKey memory olKey)
     external
     view
-    returns (MgvStructs.GlobalUnpacked memory global, MgvStructs.LocalUnpacked memory local);
+    returns (MgvStructs.GlobalUnpacked memory _global, MgvStructs.LocalUnpacked memory _local);
```
Comment (espendk) [inline src/IMangrove.sol:41]: Again, why the underscores?

**Review by espendk — COMMENTED** [review]
Great!

One question: What do we win by writing tests against the interface when the ABI is checked?

File: src/MgvRoot.sol (hunk 1 above)
Comment (lnist) [reply src/MgvRoot.sol:141]: We have _global in other function signatures as well. My editor doesn't like just "global" and I thought that was why it was introduced elsewhere, so I aligned it. It is not super important whether we go with or without _, but I think we should be consistent.

File: src/IMangrove.sol (hunk 2 above)
Comment (lnist) [reply src/IMangrove.sol:41]: This is the same signature as for your other comment. So they are just identical :)

"config" was the one with already existing _global.

File: src/MgvRoot.sol (hunk 1 above)
Comment (espendk) [reply src/MgvRoot.sol:141]: Agree on the consistency. Unless there's a reason to have the `_` I think we should avoid it.

Comment (lnist) [reply src/MgvRoot.sol:141]: global is a reserved keyword. Agreed with @adhusson to keep _.

**Comment by lnist** [comment]
Writing tests against IMangrove means being able to navigate to natspec.

**Pull request merged**